### PR TITLE
Add support for TRY CAST to the parser + transformer

### PR DESCRIPTION
**Pull request by Mytherin**
Syntax is identical to cast:

```sql
SELECT TRY_CAST('hello' as INTEGER)
```

The idea is that this returns NULL instead of throwing an exception. Note that this is only supported in the parser for now; not actually usable yet in the main system.

### CODE REVIEW

**Comment by Mytherin** [comment]
@mbasmanova

**Comment by mbasmanova** [comment]
CC: @beroyfb @pedroerp

**Comment by mbasmanova** [comment]
@Mytherin Thank you, Mark.

**Comment by beroyfb** [comment]
Thanks for adding this.

File: third_party/libpg_query/include/parser/gram.hpp [hunk 1]
```diff
@@ -925,65 +926,66 @@
...
+#define XMLTABLE 720
+#define YEAR_P 721
+#define YEARS_P 722
+#define YES_P 723
+#define ZONE 724
+#define NOT_LA 725
+#define NULLS_LA 726
+#define WITH_LA 727
+#define POSTFIXOP 728
+#define UMINUS 729
```
Comment (EthanDBer) [inline third_party/libpg_query/include/parser/gram.hpp:988]: Should this be replaced by `constexpr`?

**Pull request merged**